### ISSUE TITLE
Regularization Loss is getting added twice if model is encapsulated in another model

### ISSUE THREAD

**Issue by ksachdeva** [keras-team/keras#15909]
This is a re-submission of the issue from here https://github.com/tensorflow/tensorflow/issues/43995

As the bug reports show the issue has been reproduced by many other people including engineers from Google. 

**Description**

This pattern of creating simple models (modules) and then assembling them as part of the big model is widely used and as shown in the above code snippets and its outcome this ends up adding the regularization loss twice.

I noticed it while adding KLDivergenceRegularization from tensorflow_probability and then reproduced it using a simple example in the notebooks below

**Notebook to reproduce the issue**
https://colab.research.google.com/drive/1tPLQgjX-9IqCeUSpG5xRfuIMAv7cPSNK?usp=sharing

**Comment by qlzh727** [comment]
Triage notes: we had similar bug before, and should look into the root cause for this.

**Comment by haifeng-jin** [comment]
As we tested the actual value is not added twice. Refer to [this colab notebook](https://colab.research.google.com/github/haifeng-jin/Colabs/blob/main/keras_issue_15909.ipynb).

The multiple losses were displayed because it was printed at a premature stage.
When the model gets called at least once, the multiple loss entries will be clean up and only one loss remains.

We will not fix this unless it blocks some very important workflows because the required change is none trivial.

Thanks again for submitting the issue!
